src/WebClient.spec.js

```diff
@@ -500,15 +501,95 @@ describe('WebClient', function () {
   });
 
   describe('has rate limit handling', function () {
-    it('should expose retry headers in the response');
+    it('should expose retry headers in the response', function() {
+      const scope = nock('https://slack.com')
+        .post(/api/)
+        .reply(200, { ok: true }, {
+          'retry-after': 1
+        });
+      const client = new WebClient(token);
+      return client.apiCall('method')
+        .then((result) => {
+          assert.deepNestedInclude(result, { 'retryAfter': 1000 });
+          scope.done();
+        });
+    });
+
     // NOTE: see retry policy note below
-    it('should allow rate limit triggered retries to be turned off');
+    it('should allow rate limit triggered retries to be turned off', function() {
+      const scope = nock('https://slack.com')
+        .post(/api/)
+        .reply(429, { ok: false }, {
+          'retry-after': 0
+        })
+        .post(/api/)
+        .reply(200, { ok: true });
+      const client = new WebClient(token, { retryConfig: { retries: 0 } });
+      return client.apiCall('method')
+        .then(() => {
+          assert.fail(0, 1, 'Rate Limit retry not turned off');
+        })
+        .catch((error) => {
+          assert.isFalse(scope.isDone());
+        });
+    });
+
+
 
     describe('when a request fails due to rate-limiting', function () {
       // NOTE: is this retrying configurable with the retry policy? is it subject to the request concurrency?
-      it('should automatically retry the request after the specified timeout');
-      it('should pause the remaining requests in queue');
-      it('should emit a rate_limited event on the client');
+      it('should automatically retry the request after the specified timeout', function() {
+        const scope = nock('https://slack.com')
+          .post(/api/)
+          .reply(429, { ok: true }, {
```

Comment:
probably not `{ ok: true }`

Reply:
Will change this and every instance thereof